app/helpers/issue_view_columns_issues_helper.rb

```diff
@@ -6,49 +6,238 @@ def render_descendants_tree(issue)
+          issues_with_subissues.concat(subissues_with)
+          issues_with_subissues.concat(subissues_without)
+        else
+          issues_without_subissues << render_issue_row.call(child, level, child_hidden) if child.parent_id == parent.id
+        end
+      end
+
+      return issues_with_subissues, issues_without_subissues
+    end
+
+    # Start rendering from the top-level issue
+    issues_with_subissues, issues_without_subissues = render_issue_with_descendants.call(issue, 0)
+
+    # Append the rendered issues to the field values
+    field_values << issues_with_subissues.join("").html_safe
+    field_values << issues_without_subissues.join("").html_safe
+  end
+
+  def render_issues_default(issue, render_issue_row, collapsed_ids, columns_list, field_values)
+    render_issue_with_descendants = lambda do |parent, level, hidden = false|
+      issues = []
+
+      # Get direct descendants and sort them
+      direct_descendants = parent.descendants.select { |descendant| descendant.parent_id == parent.id }
+      sorted_issues = sort_issues(direct_descendants, columns_list)
+
+      # Traverse sorted issues recursevely
+      sorted_issues.each do |child|
+        next if (child.closed? && !issue_columns_with_closed_issues?)
+
+        child_hidden = hidden || collapsed_ids.include?(child.id)
+
+        issues << render_issue_row.call(child, level, hidden)
+        subissues = render_issue_with_descendants.call(child, level + 1, child_hidden)
+        issues.concat(subissues)
+      end
+
+      issues
+    end
+
+    # Start rendering from the root issue
+    rendered_issues = render_issue_with_descendants.call(issue, 0, false)
+    field_values << rendered_issues.join("").html_safe
+  end
+
+  def sort_issues(issues, columns_list)
+    columns_sorting_setting = RedmineIssueViewColumns.setting(:columns_sorting)
+    return issues unless columns_sorting_setting.present?
+
+    # Build sorting criteria as an array of hashes with keys :column_name and :direction
+    sorting_criteria = columns_sorting_setting.split(",").map do |column_setting|
+      column_name, direction = column_setting.split(":").map(&:strip)
+      { column_name: caption_to_name(column_name, columns_list).downcase, direction: direction }
+    end
+
+    # Use the extracted comparison lambda
+    sorted_issues = issues.to_a.sort(&comparison_lambda(sorting_criteria))
+
+    sorted_issues
+  end
+
+  # Define a method for comparison lambda
+  def comparison_lambda(sorting_criteria)
+    lambda do |a, b|
+      sorting_criteria.each do |criterion|
+        column_name = criterion[:column_name]
+        direction = criterion[:direction] == "ASC" ? 1 : -1
+
+        if column_name.start_with?("cf_")
+          # Handle custom fields
+          cf_id = column_name.sub(/^cf_/, "")
+          a_value = CustomValue.where(customized_id: a.id, customized_type: "Issue", custom_field_id: cf_id).first&.value
```

Comment:
Can't you get the info from issue without searching in the database? I remember having a code that gets a value from issue base on a field (where field is either normal field or customfield) meaning you don't need to diferentaite here between `cf_` or some thing, it just uses redmine to return the value.